src/controller/artifact/annotation/checker/annotation.go

```diff
@@ -0,0 +1,86 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checker
+
+import (
+	"fmt"
+
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/pkg/artifact"
+)
+
+const (
+	// XIcon is icon content type image/x-icon
+	XIcon = "image/x-icon"
+	// GIF is icon content type image/gif
+	GIF = "image/gif"
+	// PNG is icon content type image/png
+	PNG = "image/png"
+	// JPEG is icon content type image/jpeg
+	JPEG = "image/jpeg"
+
+	// AnnotationPrefix is the prefix of annotation
+	AnnotationPrefix = "io.goharbor.artifact"
+
+	// SkipList is the key word of skip-list annotation
+	SkipList = "skip-list"
+	// Icon is the key word of icon annotation
+	Icon = "icon"
+)
+
+var (
+	// Registry for registered annotation checkers
+	Registry = map[string]AnnotationChecker{}
+
+	// SortedAnnotationVersionList define the order of AnnotationChecker from high to low version.
+	// High version annotation checker will process annotation first.
+	// In the future we have v1alpha2 version, SortedAnnotationVersionList can be defined like
+	// SortedAnnotationVersionList = []string{V1alpha2, V1alpha1}
+	SortedAnnotationVersionList = []string{V1alpha1}
+)
+
+func init() {
+	if err := RegisterV1Alpha1Checker(); err != nil {
```

Comment:
define init() in the RegisterV1Alpha1Checker and RegisterV1Alpha2Checker? and import with `_`?